File: scripting/jvmMain/java/com/dropbear/components/LabelNative.java
package com.dropbear.components;

import com.dropbear.EucalyptusCoreLoader;

public class LabelNative {
    static {
        new EucalyptusCoreLoader().ensureLoaded();
    }

    public static native boolean labelExistsForEntity(long worldHandle, long entityId);
}

File: scripting/nativeMain/kotlin/com/dropbear/components/Label.native.kt
@file:OptIn(ExperimentalForeignApi::class)

package com.dropbear.components

import com.dropbear.DropbearEngine
import com.dropbear.EntityId
import com.dropbear.ffi.generated.*
import kotlin.String
import kotlinx.cinterop.*

internal actual fun labelExistsForEntity(entityId: EntityId): Boolean = memScoped {
    val world = DropbearEngine.native.worldHandle ?: return@memScoped false
    val out = alloc<BooleanVar>()
    dropbear_entity_label_exists_for_entity(world, entityId.raw.toULong(), out.ptr)
    out.value
}